.github/workflows/r.yml

```diff
@@ -52,6 +52,66 @@ env:
   DOCKER_VOLUME_PREFIX: ".docker/"
 
 jobs:
+  ubuntu-cpp-versions:
+    name: Arrow C++ Versions (${{ matrix.cpp_version }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # 14.0.2 apt binaries are missing but contain only minor changes
+        cpp_version: ["15.0.0" , "14.0.1",  "13.0.0"]
```

Comment:
Where do we remind ourselves to add a new version to this list, after a release?

When do we decide to drop old versions? Provided that this CI job isn't too expensive, seems fine to keep 13.0.0 on there indefinitely, there's no particular maintenance burden associated with it (the only check I see is for < 15.0.0). Since it's in a matrix build, it may be effectively free (to us) to let this list grow.

Is there a reason we can't build on 12.0, or 11.0 for that matter?

Reply:
Does this also make assumptions that patch releases are ABI-compatible with the major releases? That's probably usually true.

Reply:
> Since it's in a matrix build, it may be effectively free (to us) to let this list grow.

Eh technically true but a bit wasteful in regards to community resources (tragedy of the commons etc.) imo. At least as long as there is no specific argument that we actually need to test against all old versions vs. only against the minimum supported version + latest stable or something (see comment above, have no idea if this is the case)

> patch releases are ABI-compatible with the major releases?

Otherwise they could not be patch releases (at least in theory...).

Reply:
> Where do we remind ourselves to add a new version to this list, after a release?

We can make it part of the post release scripts.

Reply:
> At least as long as there is no specific argument that we actually need to test against all old versions vs. only against the minimum supported version + latest stable or something

I'd think you'd want to test against each version you have a conditional check for.

Hopefully these are cheap jobs to run since you don't have to compile libarrow in them. Could always move them to nightly to save compute if that's a concern, as we do for testing against older R versions.

Reply:
@nealrichardson 
> I'd think you'd want to test against each version you have a conditional check for.

That makes sense, so in this case test against 13.0.0 as the minimum version, 15.0.0 due to the #if and latest stable?

> Hopefully these are cheap jobs to run

They are pretty quick with ~7 min, maybe I am a bit to zealous in keep total ci time low (after all checking 10 version would be one of the other C++ jobs with a cache miss...). But having this discussion and documenting what we support and test is beneficial in any case :)

@paleolimbot 
> Where does that happen? I imagine I should do it in this PR?

The version bump script is run after the release and binary upload and matches thematically: `dev/release/post-11-bump-versions.sh` unless @raulcd disagrees?